experiments/PlaywrightSharp/PlaywrightSharp/ApiSurface/IPage.cs

```diff
@@ -0,0 +1,629 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) Microsoft Corporation.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *
+ * ------------------------------------------------------------------------------ 
+ * <auto-generated> 
+ * This code was generated by a tool at:
+ * /utils/doclint/generateDotnetApi.js
+ * 
+ * Changes to this file may cause incorrect behavior and will be lost if 
+ * the code is regenerated. 
+ * </auto-generated> 
+ * ------------------------------------------------------------------------------
+ */
+using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.IO;
+using System.Runtime.Serialization;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace PlaywrightSharp
+{
+    /// <summary>
+	/// <seealso cref="[EventEmitter]"/>
+	/// Page provides methods to interact with a single tab in a <see cref="IBrowser"/>, or an <a href="https://developer.chrome.com/extensions/background_pages">extension background page</a> in
+	/// Chromium. One <see cref="IBrowser"/> instance might have multiple <see cref="IPage"/> instances.
+	/// This example creates a page, navigates it to a URL, and then saves a screenshot:
+	/// The Page class emits various events (described below) which can be handled using any of Node's native 
+	/// <a href="https://nodejs.org/api/events.html#events_class_eventemitter">`EventEmitter`</a> methods, such as `on`, `once` or
+	/// `removeListener`.
+	/// This example logs a message for a single page `load` event:
+	/// To unsubscribe from events use the `removeListener` method:
+	/// </summary>
+	public interface IPage
+	{
+		event EventHandler<IPage> OnClose;
+		event EventHandler<IConsoleMessage> OnConsole;
```

Comment:
I think we will need to auto-generate EvenHandles to wrap up the values.
Here, I would expect a CloseEvenHandler with a ConsoleMessage Property.

I think we talked about removing the `On`.